spec/requests/registrations_spec.rb

```diff
@@ -0,0 +1,91 @@
+require 'rails_helper'
+
+RSpec.describe 'Registrations', type: :request do
+  describe 'POST #create' do
+    subject(:request) { post '/signup.json', params: params }
+
+    let(:response_body) { JSON.parse(response.body).deep_symbolize_keys }
+
+    context 'without logged in user', :aggregate_failures do
+      context 'with valid params' do
+        let(:params) { { user: attributes_for(:user, full_name: 'Mary Jane') } }
+
+        let(:new_user) { User.last }
+
+        let(:expected_body) do
+          {
+            active: false,
+            address: new_user.address,
+            admin: false,
+            cellphone: new_user.cellphone,
+            companies: [],
+            cpf: new_user.cpf,
+            created_at: new_user.created_at.iso8601(3),
+            email: new_user.email,
+            full_name: new_user.full_name,
+            id: new_user.id,
+            picture_url: nil,
+            self_registered: true,
+            updated_at: new_user.updated_at.iso8601(3)
+          }
+        end
+
+        it 'shows success message' do
+          request
+
+          expect(response_body).to eq expected_body
+          expect(response).to be_successful
+        end
+
+        it 'creates a new record in the database' do
+          expect { request }.to change(User, :count).by(1)
+        end
+
+        it 'sends an email to the admins', :aggregate_failures do
+          expect { request }.to have_enqueued_job(ActionMailer::MailDeliveryJob)
```

Comment:
Here you are checking if **any** email is enqueued, so if you change the called mailer, this test will not fail.

To avoid this, you can do something like:
```ruby
expect { request }.to have_enqueued_mail(AdminMailer, :user_self_registration_notification)
```